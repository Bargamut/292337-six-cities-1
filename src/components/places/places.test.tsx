import * as React from 'react';
import * as ShallowRenderer from 'react-test-renderer/shallow';

import Places from './places';

const renderer = new ShallowRenderer();

const mocks = {
  citiesPlaces: [
    {
      id: 0,
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 0,
          longitude: 0,
          zoom: 13
        }
      },
      type: `Apartment`,
      previewImage: ``,
      images: [`1.jpg`, `2.jpg`, `3.jpg`],
      bedrooms: 1,
      maxAdults: 2,
      goods: [ `Towels`, `Dishwasher`, `Breakfast` ],
      host: {
        id: 25,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: ``,
      isPremium: true,
      isFavorite: false,
      title: `Amsterdam Apartment Premium`,
      price: 0,
      rating: 0,
      location: {
        latitude: 0,
        longitude: 0,
        zoom: 13
      }
    },
    {
      id: 1,
      city: {
        name: `Berlin`,
        location: {
          latitude: 0,
          longitude: 0,
          zoom: 13
        }
      },
      type: `Private room`,
      previewImage: ``,
      images: [`1.jpg`, `2.jpg`, `3.jpg`],
      bedrooms: 1,
      maxAdults: 2,
      goods: [ `Towels`, `Dishwasher`, `Breakfast` ],
      host: {
        id: 25,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: ``,
      isPremium: false,
      isFavorite: false,
      title: `Berlin Private room`,
      price: 0,
      rating: 0,
      location: {
        latitude: 0,
        longitude: 0,
        zoom: 13
      }
    },
    {
      id: 2,
      city: {
        name: `Sydney`,
        location: {
          latitude: 0,
          longitude: 0,
          zoom: 13
        }
      },
      type: `Apartment`,
      previewImage: ``,
      images: [`1.jpg`, `2.jpg`, `3.jpg`],
      bedrooms: 1,
      maxAdults: 2,
      goods: [ `Towels`, `Dishwasher`, `Breakfast` ],
      host: {
        id: 25,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: ``,
      isPremium: false,
      isFavorite: false,
      title: `Sydney Apartment`,
      price: 0,
      rating: 0,
      location: {
        latitude: 0,
        longitude: 0,
        zoom: 13
      }
    },
    {
      id: 3,
      city: {
        name: `Saint-Petersburg`,
        location: {
          latitude: 0,
          longitude: 0,
          zoom: 13
        }
      },
      type: `Apartment`,
      previewImage: ``,
      images: [`1.jpg`, `2.jpg`, `3.jpg`],
      bedrooms: 1,
      maxAdults: 2,
      goods: [ `Towels`, `Dishwasher`, `Breakfast` ],
      host: {
        id: 25,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: ``,
      isPremium: true,
      isFavorite: false,
      title: `Saint-Petersburg Apartment Premium`,
      price: 0,
      rating: 0,
      location: {
        latitude: 0,
        longitude: 0,
        zoom: 13
      }
    },
    {
      id: 4,
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 0,
          longitude: 0,
          zoom: 13
        }
      },
      type: `Private room`,
      previewImage: ``,
      images: [`1.jpg`, `2.jpg`, `3.jpg`],
      bedrooms: 1,
      maxAdults: 2,
      goods: [ `Towels`, `Dishwasher`, `Breakfast` ],
      host: {
        id: 25,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: ``,
      isPremium: true,
      isFavorite: false,
      title: `Amsterdam Private room Premium`,
      price: 0,
      rating: 0,
      location: {
        latitude: 0,
        longitude: 0,
        zoom: 13
      }
    }
  ],
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 51.23,
      longitude: 51.24,
      zoom: 13
    }
  }
};

it(`Places correctly renders`, () => {
  const {citiesPlaces, city} = mocks;

  const places = renderer.render(
    <Places
      citiesPlaces={citiesPlaces}
      city={city}
      activeItem={citiesPlaces[0]}
      onChangeOffersFilter={jest.fn()}
      onSetActiveItem={jest.fn()}
    />
  );

  expect(places).toMatchSnapshot();
});
