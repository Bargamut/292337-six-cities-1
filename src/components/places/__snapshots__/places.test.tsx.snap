// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Places correctly renders 1`] = `
<div
  className="cities__places-container container"
>
  <section
    className="cities__places places"
  >
    <h2
      className="visually-hidden"
    >
      Places
    </h2>
    <b
      className="places__found"
    >
      5
       
      places
       to stay in 
      Amsterdam
    </b>
    <PlacesSorting
      onChangeOffersFilter={[MockFunction]}
      sortItems={
        Array [
          Object {
            "caption": "Popular",
            "value": "popular",
          },
          Object {
            "caption": "Price: low to high",
            "order": "ASC",
            "value": "price",
          },
          Object {
            "caption": "Price: high to low",
            "order": "DESC",
            "value": "price",
          },
          Object {
            "caption": "Top rated first",
            "order": "DESC",
            "value": "rating",
          },
        ]
      }
    />
    <PlacesList
      citiesPlaces={
        Array [
          Object {
            "bedrooms": 1,
            "city": Object {
              "location": Object {
                "latitude": 0,
                "longitude": 0,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "",
            "goods": Array [
              "Towels",
              "Dishwasher",
              "Breakfast",
            ],
            "host": Object {
              "avatarUrl": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 0,
            "images": Array [
              "1.jpg",
              "2.jpg",
              "3.jpg",
            ],
            "isFavorite": false,
            "isPremium": true,
            "location": Object {
              "latitude": 0,
              "longitude": 0,
              "zoom": 13,
            },
            "maxAdults": 2,
            "previewImage": "",
            "price": 0,
            "rating": 0,
            "title": "Amsterdam Apartment Premium",
            "type": "Apartment",
          },
          Object {
            "bedrooms": 1,
            "city": Object {
              "location": Object {
                "latitude": 0,
                "longitude": 0,
                "zoom": 13,
              },
              "name": "Berlin",
            },
            "description": "",
            "goods": Array [
              "Towels",
              "Dishwasher",
              "Breakfast",
            ],
            "host": Object {
              "avatarUrl": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 1,
            "images": Array [
              "1.jpg",
              "2.jpg",
              "3.jpg",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 0,
              "longitude": 0,
              "zoom": 13,
            },
            "maxAdults": 2,
            "previewImage": "",
            "price": 0,
            "rating": 0,
            "title": "Berlin Private room",
            "type": "Private room",
          },
          Object {
            "bedrooms": 1,
            "city": Object {
              "location": Object {
                "latitude": 0,
                "longitude": 0,
                "zoom": 13,
              },
              "name": "Sydney",
            },
            "description": "",
            "goods": Array [
              "Towels",
              "Dishwasher",
              "Breakfast",
            ],
            "host": Object {
              "avatarUrl": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 2,
            "images": Array [
              "1.jpg",
              "2.jpg",
              "3.jpg",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 0,
              "longitude": 0,
              "zoom": 13,
            },
            "maxAdults": 2,
            "previewImage": "",
            "price": 0,
            "rating": 0,
            "title": "Sydney Apartment",
            "type": "Apartment",
          },
          Object {
            "bedrooms": 1,
            "city": Object {
              "location": Object {
                "latitude": 0,
                "longitude": 0,
                "zoom": 13,
              },
              "name": "Saint-Petersburg",
            },
            "description": "",
            "goods": Array [
              "Towels",
              "Dishwasher",
              "Breakfast",
            ],
            "host": Object {
              "avatarUrl": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 3,
            "images": Array [
              "1.jpg",
              "2.jpg",
              "3.jpg",
            ],
            "isFavorite": false,
            "isPremium": true,
            "location": Object {
              "latitude": 0,
              "longitude": 0,
              "zoom": 13,
            },
            "maxAdults": 2,
            "previewImage": "",
            "price": 0,
            "rating": 0,
            "title": "Saint-Petersburg Apartment Premium",
            "type": "Apartment",
          },
          Object {
            "bedrooms": 1,
            "city": Object {
              "location": Object {
                "latitude": 0,
                "longitude": 0,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "",
            "goods": Array [
              "Towels",
              "Dishwasher",
              "Breakfast",
            ],
            "host": Object {
              "avatarUrl": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 4,
            "images": Array [
              "1.jpg",
              "2.jpg",
              "3.jpg",
            ],
            "isFavorite": false,
            "isPremium": true,
            "location": Object {
              "latitude": 0,
              "longitude": 0,
              "zoom": 13,
            },
            "maxAdults": 2,
            "previewImage": "",
            "price": 0,
            "rating": 0,
            "title": "Amsterdam Private room Premium",
            "type": "Private room",
          },
        ]
      }
      onClickImageItem={[MockFunction]}
    />
  </section>
  <div
    className="cities__right-section"
  >
    <CitiesMap
      hasActivePoint={true}
      location={
        Object {
          "latitude": 0,
          "longitude": 0,
          "zoom": 13,
        }
      }
      placesCoords={
        Array [
          Object {
            "latitude": 0,
            "longitude": 0,
            "zoom": 13,
          },
          Object {
            "latitude": 0,
            "longitude": 0,
            "zoom": 13,
          },
          Object {
            "latitude": 0,
            "longitude": 0,
            "zoom": 13,
          },
          Object {
            "latitude": 0,
            "longitude": 0,
            "zoom": 13,
          },
          Object {
            "latitude": 0,
            "longitude": 0,
            "zoom": 13,
          },
        ]
      }
    />
  </div>
</div>
`;
